avl_tree/AvlTree.java

```diff
@@ -0,0 +1,306 @@
+import java.util.Scanner;
+import java.util.NoSuchElementException;
+import java.util.InputMismatchException;
+
+public class AvlTree {
+
+	public static void main(String[] args) {
+		AVL avl = new AVL();
+		Scanner inp = new Scanner(System.in);
+		while (true) {
+			try {
+				System.out.println("\nTo insert an element Enter 1");
+				System.out.println("To delete an element Enter 2");
+				System.out.println("To print AVL tree preorder Enter 3");
+				System.out.println("To EXIT Enter 4");
+				int a = inp.nextInt();
+				switch (a) {
+				case 1:
+					int in = inp.nextInt();
+					avl.insert(in);
+					break;
+				case 2:
+					int d = inp.nextInt();
+					avl.del(d);
+					break;
+				case 3:
+					if (avl.isEmpty()) {
+						System.out.println("null");
+					} else {
+						avl.prntIn(avl.root);
+						System.out.println(" ");
+					}
+					break;
+				case 4:
+					break;
+				default:
+					System.out.println("Sorry input number is not valid ");
+					break;
+				}
+				if (a == 4) {
+					break;
+				}
+			} catch (InputMismatchException e) {
+				System.out.println("Input not valid!!!\nplease Integers numbers only");
+				break;
+			}
+
+		}
+	}
+}
+
+class AVL {
+	public NodeAVL root = null;
+
+	public void insert(int data) {
+		// call these method when you wanna insert an element in the tree;
+		if (root == null) {
+			root = new NodeAVL(data);
+		} else
+			root = insert(root, data);
+	}
+
+	private NodeAVL insert(NodeAVL node, int data) {

```

Comment:
It's great that you are using the concept of Operator overloading for these `insert` methods but it creates ambiguity for someone who is reading the code. 
Also I don't think two methods are required here. One single method with signature `NodeAVL insert(NodeAVL, int)` will also work.